src/main/kotlin/graphql/kickstart/tools/resolver/MethodFieldResolver.kt

```diff
@@ -94,7 +96,11 @@ internal class MethodFieldResolver(
         if (this.additionalLastArgument) {
             when (this.method.parameterTypes.last()) {
                 null -> throw ResolverError("Expected at least one argument but got none, this is most likely a bug with graphql-java-tools")
-                options.contextClass -> args.add { environment -> environment.getContext() }
+                options.contextClass -> args.add { environment ->
+                    environment.graphQlContext[options.contextClass]
```

Comment:
Let's print a warning whenever the right part of the `?:` operator gets hit.

This is how I would word it roughly:

```
Generic context class has been deprecated by graphql-java. To continue using a custom context class as the last parameter in resolver methods please insert it into the new GraphQLContext class when building the ExecutionInput.
```